impl/client/src/main/javascript/web/pentaho/lang/EventSource.js

```diff
@@ -569,21 +591,44 @@ define([
 
           if(isCanceled && isCanceled(event)) return null;
         }
-
       }
-
     } finally {
       queue.emittingLevel--;
     }
 
     return event;
   }
 
+  /**
+   * Determines if a given event object is canceled.
+   *
+   * @param {!pentaho.lang.Event} event - The event object.
+   *
+   * @return {boolean} `true` if it is canceled; `false`, otherwise.
+   *
+   * @private
+   */
   function event_isCanceled(event) {
     return event.isCanceled;
   }
 
-  function errorHandlerLog(ex, event, type, phase) {
-    logger.log("Event listener of '" + type + ":" + phase + "' failed. \nStack: " + ex.stack);
+  // TODO: check log level!
+
+  /**
+   * Logs an error thrown by an event listener.
+   *
+   * @param {any} error - The thrown value.
```

Comment:
Unfortunately, JavaScript allows throwing any type of value and we don't control what listener functions decide to throw...